lib/code_fund/stats/user_impressions.ex

```diff
@@ -0,0 +1,103 @@
+defmodule CodeFund.Stats.UserImpressions do
+  use GenServer
+  use CodeFundWeb, :query
+  alias CodeFund.Schema.UserImpression
+
+  @refresh_interval :timer.minutes(60)
+
+  defmodule State do
+    defstruct impression_count: 0, click_count: 0, click_rate: 0.0, distribution_amount: 0.0
+  end
+
+  # Client API
+
+  def last_thirty_days() do
+    GenServer.call(__MODULE__, :stats)
+  end
+
+  # Invoked by the Supervisor process
+  def start_link do
+    GenServer.start_link(__MODULE__, %State{}, name: __MODULE__)
+  end
+
+  # Inside the spawned GenServer process
+  def init(state) do
```

Comment:
This is a good idea (in spite of and regardless of heroku's use) since it can respin the stats on a node startup and no state will get lost that needs to get written permanently.